### ISSUE TITLE
iOS: onError (AVFoundationErrorDomain) being thrown when changing selectedTextTrack

### ISSUE THREAD

**Issue by brycnguyen** [TheWidlarzGroup/react-native-video#3480]
# Bug
Originally I had text track selection working fine in https://github.com/react-native-video/react-native-video/commit/8f1bdb7c36cb3b42925312b9a9952432f4735ba5. One of the other commits in the release is causing this to break.

{"error": {"code": -11800, "domain": "AVFoundationErrorDomain", "localizedDescription": "The operation could not be completed", "localizedFailureReason": "An unknown error occurred (-12783)", "localizedRecoverySuggestion": ""}, "target": 3557}

Being thrown for onError whenever I switch selected text track

## Platform
Which player are you experiencing the problem on:
* iOS

## Environment info
Library version: 6.0.0-beta.4
Device: iPhone 14 simulator

## Steps To Reproduce
1. Pass in an array of text tracks to react-native-video
2. Change the selectedTextTrack
3. onError throws error mentioned in the ticket
4. Video does not play

## Expected behaviour
1. Changing the selected text track should continue to play the video

**Comment by gkueny** [comment]
I tried to check this issue, but without any luck for now.

However, if it helps others, here's a minimal repo to reproduce the problem https://github.com/gkueny/VideoErrorThrownOnSelectedTrackReproduction


@brycnguyen with my minimal reproduction mentioned issue is still present when I use `yarn add react-native-video@https://github.com/react-native-video/react-native-video#f1bdb7c36cb3b42925312b9a9952432f4735ba5` (https://github.com/react-native-video/react-native-video/commit/8f1bdb7c36cb3b42925312b9a9952432f4735ba5)

> Originally I had text track selection working fine

did you use the commit in question?
Or just via a patch on one of the versions you use? (& if so which one ?)

**Comment by KrzysztofMoch** [comment]
@gkueny Thank you for sharing repo ❤️
@brycnguyen I have tried v6 beta 4 & 5 and it's not working - not sure what is the issue, because error message is not show any readable info 🫠

side loading subtitles is not liked by apple (there is no official api to do it 😐)

I will try to make some investigation but I am not promise that i will make this work

**Comment by gkueny** [comment]
@KrzysztofMoch I think I have a guess but not sure for now. 
But it's definitely an issue between https://github.com/react-native-video/react-native-video/commit/8f1bdb7c36cb3b42925312b9a9952432f4735ba5 & https://github.com/react-native-video/react-native-video/commit/5aaa53d8b80f6bac1d48acb863bd449fb3c15b41

I was able to make it work with `beta.3` & by manually add 

```diff
- for i in 0 ..< trackCount where (player?.currentItem?.tracks[i].assetTrack?.hasMediaCharacteristic(.legible)) != nil {
+ for i in 0 ..< trackCount where player?.currentItem?.tracks[i].assetTrack?.hasMediaCharacteristic(.legible) ?? false {
```

& do some tweak to

```diff
- "textTracks": textTracks.map(\.json),
+ "textTracks": self._textTracks?.compactMap { $0.json } ?? textTracks.map(\.json),
```

It was late & I do not remember the real solution. But I will check again

**Comment by gkueny** [comment]
@KrzysztofMoch my bad, I went in the wrong direction, you can ignore my comment

> I was able to make it work with beta.3 & by manually add [...]

**Comment by KrzysztofMoch** [comment]
I have tested fix made by @gkueny and it worked for me - will be included in 6.0.0-beta.6
Please reopen if it will not work for you
